pkg/azure/aks/price_store.go

```diff
@@ -106,34 +94,17 @@ func (p *PriceStore) buildListOptions(locationList []string) *retailPriceSdk.Ret
 	}
 }
 
-func (p *PriceStore) determineMachineOperatingSystem(sku retailPriceSdk.ResourceSKU) MachineOperatingSystem {
-	switch {
-	case strings.Contains(sku.ProductName, "Windows"):
-		return Windows
-	default:
-		return Linux
-	}
-}
-
-func (p *PriceStore) determineMachinePriority(sku retailPriceSdk.ResourceSKU) MachinePriority {
-	switch {
-	case strings.Contains(sku.SkuName, "Spot"):
-		return Spot
-	default:
-		return OnDemand
-	}
-
-}
-
 func (p *PriceStore) PopulatePriceStore(locationList []string) error {
 	startTime := time.Now()
-	p.logger.LogAttrs(p.context, slog.LevelInfo, "populating price map")
+	p.logger.Info("populating price store")
 
-	p.lock.Lock()
-	defer p.lock.Unlock()
+	p.regionMapLock.Lock()
+	defer p.regionMapLock.Unlock()
 
-	pager := p.retailPriceClient.NewListPager(p.buildListOptions(locationList))
+	// clear the existing region map
+	p.RegionMap = make(map[string]PriceByPriority)
 
+	pager := p.retailPriceClient.NewListPager((p.buildListOptions(locationList)))
```

Comment:
Linter is yelling here due to too many parenthesis:

```suggestion
	pager := p.retailPriceClient.NewListPager(p.buildListOptions(locationList))
```